arch/arc/core/cache.c

```diff
@@ -110,36 +94,37 @@ static void dcache_flush_mlines(u32_t start_addr, u32_t size)
 				break;
 			}
 		} while (1);
-		start_addr += DCACHE_LINE_SIZE;
-	} while (start_addr <= end_addr);
+		*((u32_t *)start_addr) += DCACHE_LINE_SIZE;
```

Comment:
This is still scribbling over the memory area we're trying to flush.

Reply:
Fixed this.